src/linttest/regression_test.go

```diff
@@ -1243,3 +1243,31 @@ function f($x) {
 }
 `)
 }
+
+func TestIssue406(t *testing.T) {
```

Comment:
Regression tests are mostly for fixes of the old code.
New checks are considered to be a new feature and their tests usually go in other files (almost any file will do).

The main purpose of regression test is that even if it looks like it's excessive, it should never be removed as it's bound to the bug fix. Contrary to that, tests from other files can be removed if they become unnecessary.